typed-racket-lib/typed-racket/optimizer/logging.rkt

```diff
@@ -90,25 +92,20 @@
 ;;--------------------------------------------------------------------
 
 (define (line+col->string stx)
-  (let ([line (syntax-line stx)]
-        [col  (syntax-column stx)])
-    (if (and line col)
-        (format "~a:~a" line col)
-        "(no location)")))
+  (define line (syntax-line stx))
+  (define col (syntax-column stx))
+  (if (and line col) (format "~a:~a" line col) "(no location)"))
 
 (define (format-irritants irritants)
   (if (null? irritants)
       ""
       (format " -- caused by: ~a"
               (string-join
-               (map (lambda (irritant)
-                      (let ([irritant (locate-stx irritant)])
-                        (format "~a ~s"
-                                (line+col->string irritant)
-                                (syntax->datum irritant))))
-                    (sort irritants <
-                          #:key (lambda (x)
-                                  (or (syntax-position x) 0))))
+               (for/list ([irritant (in-list (sort irritants
+                                                   <
+                                                   #:key (lambda (x) (or (syntax-position x) 0))))])
+                 (let ([irritant (locate-stx irritant)])
```

Comment:
Shouldn't this become a `define`?

Reply:
No, it's shadowing the `irritant` loop variable.